utils/dockerutil.py

```diff
@@ -459,6 +463,42 @@ def image_tag_extractor(cls, entity, key):
 
         return None
 
+    def image_name_extractor(self, co):
+        """
+        Returns the image name for a container, either directly from the
+        container's Image property or by inspecting the image entity if
+        the reference is its sha256 sum and not its name.
+        Result is cached for performance, no invalidation planned as image
+        churn is low on typical hosts.
+        """
+        if "Image" in co:
+            image = co.get('Image', '')
+            if image.startswith('sha256:'):
+                # Some orchestrators setup containers with image checksum instead of image name
+                try:
+                    if image in self._image_sha_to_name_mapping:
+                        return self._image_sha_to_name_mapping[image]
+                    else:
+                        image_spec = self.client.inspect_image(image)
+                        try:
+                            name = image_spec.get('RepoTags')[0]
+                            self._image_sha_to_name_mapping[image] = name
+                            return name
+                        except Exception:
```

Comment:
This exception handling is risky. Either we expect a precise error and we should only catch that, or we catch everything with `Exception` but log something. Otherwise we open the door to sneaky bugs.